### PR TITLE
Support with clause

### PR DESCRIPTION
**Pull request by pgrivachev**
`with` clause is very often used in dbt models. I decided to add to suggestion list such tables. The issue was reported here: https://fivetran.atlassian.net/browse/RD-244313.

### CODE REVIEW

**Review by fivetran-dmitryboykov — CHANGES_REQUESTED** [review]
Just some changes.

**Review by fivetran-dmitryboykov — APPROVED** [review]
Looks good! Do we need some unit tests on `server/src/utils/ZetaSqlUtils.ts`?

**Comment by pgrivachev** [comment]
> Looks good! Do we need some unit tests on `server/src/utils/ZetaSqlUtils.ts`?

Of course, adding tests is a good idea in most cases. Added.

File: server/src/test/utils/ZetaSqlUtils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { ParseLocationRangeProto__Output } from '@fivetrandevelopers/zetasql/lib/types/zetasql/ParseLocationRangeProto';
+import { assertThat } from 'hamjest';
+import { positionInRange, rangeContainsRange, rangesEqual } from '../../utils/ZetaSqlUtils';
+
+describe('ZetaSqlUtils.spec', () => {
+  it('positionInRange should return if range contains specified position', () => {
```
Comment (fivetran-dmitryboykov) [inline server/src/test/utils/ZetaSqlUtils.spec.ts:6]: return what?

File: server/src/test/utils/ZetaSqlUtils.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    assertThat(positionInRange(5, createRange(5, 8)), true);
+    assertThat(positionInRange(7, createRange(5, 8)), true);
+    assertThat(positionInRange(8, createRange(5, 8)), true);
+
+    assertThat(positionInRange(4, createRange(5, 8)), false);
+    assertThat(positionInRange(9, createRange(5, 8)), false);
+  });
+
+  it('rangeContainsRange should return if range contains another range', () => {
```
Comment (fivetran-dmitryboykov) [inline server/src/test/utils/ZetaSqlUtils.spec.ts:15]: return what?

**Review by fivetran-dmitryboykov — APPROVED** [review]

**Review by fivetran-dmitryboykov — APPROVED** [review]

**Pull request merged**